lib/wab/impl/data.rb

```diff
@@ -375,14 +375,13 @@ def values_eql?(v0, v1)
       end
 
       def key_to_int(key)
-        return key if key.is_a?(Integer)
+        return key if (key.is_a?(Integer) || WAB::Utils.pre_24_fixnum?(key))
 
         key = key.to_s if key.is_a?(Symbol)
         if key.is_a?(String)
           i = key.to_i
           return i if i.to_s == key
         end
-        return key if WAB::Utils.pre_24_fixnum?(key)
 
         raise StandardError.new("path key must be an integer for an Array.") unless i.to_s == key
```

Comment:
since `i` is defined in the `if` block at ln#381, would it be still available for the `raise` condition?